mitiq/zne/inference.py

```diff
@@ -920,6 +920,101 @@ def reduce(self) -> float:
+    of fake nodes known as Chebyshev-Lobatto points. This method gives a better
+    interpolation when there are a lot of nodes, especially close to the
+    edges of the domain.
+
+    Reference:
+    S. De Marchi, F. Marchetti, E. Perracchione, D. Poggiali,
+    Polynomial interpolation via mapped bases without resampling link, JCAM.
+
+    Args:
+        scale_factors: Sequence of noise scale factors at which
+                       expectation values should be measured.
+        shot_list: Optional sequence of integers corresponding to the number
+                   of samples taken for each expectation value. If this
+                   argument is explicitly passed to the factory, it must have
+                   the same length of scale_factors and the executor function
+                   must accept "shots" as a valid keyword argument.
+
+    Raises:
+        ValueError: If data is not consistent with the extrapolation model.
+        ExtrapolationWarning: If the extrapolation fit is ill-conditioned.
+    """
+
+    @staticmethod
+    def extrapolate(
+        scale_factors: Sequence[float],
+        exp_values: Sequence[float],
+        full_output: bool = False,
+    ) -> Union[
+        float,
+        Tuple[
+            float,
+            Union[float, None],
+            List[float],
+            Union[np.ndarray, None],
+            Callable[[float], float],
+        ],
+    ]:
+        # Richardson extrapolation is a particular case of a polynomial fit
+        # with order equal to the number of data points minus 1.
+        order = len(scale_factors) - 1
+
+        # Mapping to the fake nodes
+        _S_scale_factors = RungeFactory._map(scale_factors)
+
+        return PolyFactory.extrapolate(
+            _S_scale_factors, exp_values, order, full_output
+        )
+
+    def reduce(self) -> float:
+        """Evaluates the zero-noise limit found by applying the fake node
+        Richardson extrapolation to the internal data stored in the factory.
+
+        Returns:
+            The zero-noise limit.
+        """
+
+        (
+            self._zne_limit,
+            self._zne_error,
+            self._opt_params,
+            self._params_cov,
+            self._zne_curve,
+        ) = self.extrapolate(  # type: ignore
+            self.get_scale_factors(),
+            self.get_expectation_values(),
+            full_output=True,
+        )
+
+        self._already_reduced = True
+        return self._zne_limit
+
+    @staticmethod
+    def _map(scale_factors):
+        """A function that maps the scale_factors to the Chebyshev-Lobatto points
+        Args:
+        scale_factors: Sequence of noise scale factors at which
+                       expectation values should be measured.
+        Returns:
+            A list of fake nodes (Chebyshev-Lobatto points)
+        """
+
+        # Define the interval [a, b] we are mapping to
+        a = 1.0
+        b = scale_factors[-1]
```

Comment:
I don't remember if `BatchedFactory`s require at least two non-unique scale factors, but if not there should be a check here that `b > a`.

Reply:
I believe `BatchedFactory` raises a value error if len(scale_factors) <2. It doesn't check if they are unique or not. So I will be adding a check.